lib/src/component_declaration/component_base.dart

```diff
@@ -444,6 +445,38 @@ abstract class UiStatefulComponent<TProps extends UiProps, TState extends UiStat
   // ----------------------------------------------------------------------
 }
 
+class _WarnOnModify<K, V> extends MapView<K, V> {
+
+  //Used to customize warning based on whether the data is props or state
+  bool isProps;
+
+  String message;
+
+  _WarnOnModify(Map componentData, bool isProps): super(componentData){
```

Comment:
Not required to change this but just to let you know, Dart Constructors are pretty smart... You can actually set class level fields without a body by doing this: `_WarnOnModify(Map componentData, bool this.isProps): super(componentData);` Cool right?